### PR TITLE
Improve Dockerfile

### PR DESCRIPTION
**Pull request by PeterDaveHello**
1. Set the shell to bash is more proper for current instructions in the Dockerfile.
2. Removal of the unused packages save the size a lot.

```
$ docker images
REPOSITORY                    TAG                 IMAGE ID            CREATED             SIZE
docker-osx                    newone              747ac72ebc3f        4 days ago          2.52GB
docker-osx                    oldone              81c8272d3156        4 days ago          3.28GB
```

3. Adjust the spaces/indents and empty lines in Dockefile

See the commit message for more details! (feel free to ping me if it's not enough)

### CODE REVIEW

**Comment by PeterDaveHello** [comment]
It'll be great if the bash unofficial strict mode (`set -euo pipefail`) can be enabled in the future to improve the robustness of the script, but there are still many refactor works need to be done first, also maybe some of the config files should be seperated from the Dockerfile to have better maintainability ;)

File: Dockerfile [hunk 1]
```diff
@@ -66,7 +65,7 @@ RUN tee -a /etc/pacman.conf <<< '[community-testing]' \
     && tee -a /etc/pacman.conf <<< 'Include = /etc/pacman.d/mirrorlist'
 
 RUN pacman -Syu --noconfirm \
-    && pacman -S sudo git make automake gcc python go autoconf cmake pkgconf alsa-utils fakeroot vim nano --noconfirm \
```
Comment (PeterDaveHello) [inline Dockerfile:69]: @sickcodes this part will need your confirmation;)

**Comment by sickcodes** [comment]
Thanks for this :) I will check and merge shortly as I'm in the middle of adding some new Dockerfiles 😇

File: Dockerfile (hunk 1 above)
Comment (PeterDaveHello) [reply Dockerfile:69]: I didn't see they are used, hope I'm not wrong 😅

Comment (sickcodes) [reply Dockerfile:69]: Everything is perfect except python is required for gibMacOS :)

**Review by sickcodes — CHANGES_REQUESTED** [review]

**Review by sickcodes — APPROVED** [review]
Python is installed later anyway. Thanks!

**Comment by sickcodes** [comment]
I added the following arguments btw:

```bash
docker build --build-arg RANKMIRRORS=yes \
    --build-arg MIRROR_COUNTRY=TW \
    -t docker-osx-dev .
```

It should build MUCH faster!

**Comment by PeterDaveHello** [comment]
Cool, something should be added, please also consider to separate some configs out of Dockerfile 😃

**Comment by sickcodes** [comment]
Added alsa-utils and openssh back into this build as they are needed during run-time: https://github.com/sickcodes/Docker-OSX/commit/2a949dfb02b89eb420311a0f756042d3c9af74f8

**Pull request merged**